### ISSUE TITLE
Move atomic functions for double out of gauge_plaq.cu

### ISSUE THREAD

**Issue by mathiaswagner** [lattice/quda#309]
The following functions are implemented in `gauge_plaq.cu`.
 ```
 static __inline__ __device__ double atomicAdd(double *addr, double val)
...
static  __inline__ __device__ double2 atomicAdd(double2 *addr, double2 val)
```

They should be moves to a header so that other parts of quda can also use them.

**Comment by mikeaclark** [comment]
There is an `atomicAdd` defined in clover_invert.cu as well.

**Comment by mathiaswagner** [comment]
I believe this has been included in the merge of #253.
